go/stats/counters_test.go

```diff
@@ -240,3 +242,29 @@ func TestCountersFuncWithMultiLabels_Hook(t *testing.T) {
 		t.Errorf("want %#v, got %#v", v, gotv)
 	}
 }
+
+func TestCountersCombineDimension(t *testing.T) {
+	clear()
+	// Empty labels shouldn't be combined.
+	c0 := NewCountersWithSingleLabel("counter_combine_dim0", "help", "")
+	c0.Add("c1", 1)
+	assert.Equal(t, `{"c1": 1}`, c0.String())
+
+	clear()
+	*combineDimensions = "a,c"
+
+	c1 := NewCountersWithSingleLabel("counter_combine_dim1", "help", "label")
+	c1.Add("c1", 1)
+	assert.Equal(t, `{"c1": 1}`, c1.String())
+
+	c2 := NewCountersWithSingleLabel("counter_combine_dim2", "help", "a")
+	c2.Add("c1", 1)
+	assert.Equal(t, `{"all": 1}`, c2.String())
+
+	c3 := NewCountersWithSingleLabel("counter_combine_dim3", "help", "a")
+	assert.Equal(t, `{"all": 0}`, c3.String())
+
+	c4 := NewCountersWithMultiLabels("counter_combine_dim4", "help", []string{"a", "b", "c"})
+	c4.Add([]string{"c1", "c2", "c3"}, 1)
+	assert.Equal(t, `{"all.c2.all": 1}`, c4.String())
```

Comment:
Can you explain how this test works? What happens to c1 and c3?

Reply:
Added comment, and improved the test:
// Anything under "a" and "c" should get reported under a consolidated "all" value
// instead of the specific supplied values.